lib/service/Service.ts

```diff
@@ -693,6 +693,29 @@ class Service extends EventEmitter {
     const nodePubKey = isNodePubKey(args.nodeIdentifier) ? args.nodeIdentifier : this.pool.resolveAlias(args.nodeIdentifier);
     return this.pool.discoverNodes(nodePubKey);
   }
+  /*
+   * Subscribe to alerts.
+   */
+  public subscribeAlerts = (
+      callback: (type: AlertType, message: string, payload: ChannelBalanceAlert) => void,
+      cancelled$: Observable<void>,
+  ) => {
+    const noBalanceObservables: Observable<ChannelBalanceAlert>[] = [];
+    this.swapClientManager.swapClients.forEach((swapClient) => {
+      noBalanceObservables.push(fromEvent<ChannelBalanceAlert>(swapClient, 'lowBalance'));
+    });
+
+    const noBalance$ = merge(...noBalanceObservables).pipe(takeUntil(cancelled$)); // cleanup listeners when cancelled$ emits a value
```

Comment:
nit: have you tried marble testing for rxjs? If you're up for it we could extract `getNoBalance$` function from here.

https://rxjs.dev/guide/testing/marble-testing

Some more examples from market-maker-tools repository:
https://github.com/ExchangeUnion/market-maker-bot/blob/master/src/opendex/orders.spec.ts

Reply:
I would like to add rxjs tests too but that's new for me and would take some time, I'm committing without it for now, maybe someone more experienced can take a look at it, otherwise in a spare time I can take a look at it again